models/benefitsRecords.model.js

```diff
@@ -16,7 +16,51 @@ module.exports = {
     }
     return await BenefitsRecordsSchema.create(benefitRecordOrRecords)
   },
-  async listBenefits() {
+  async aggregateBenefitEvents(startAt) {
+    return await BenefitsRecordsSchema.aggregate([
+      {
+        $match: {
+          created_at: {
+            $gte: startAt,
+          },
+        },
+      },
+      {
+        $group: {
+          _id: {
+            benefit_id: "$benefit_id",
+            event_type: "$event_type",
+          },
+          count: { $sum: 1 },
+        },
+      },
+      {
+        $group: {
+          _id: "$_id.benefit_id",
+          events: {
+            $push: {
+              k: "$_id.event_type",
+              v: "$count",
+            },
+          },
+        },
+      },
+      {
+        $addFields: {
+          events: {
+            $arrayToObject: "$events",
+          },
+          id: "$_id",
+        },
+      },
+      {
+        $project: {
+          _id: 0,
+        },
+      },
+    ])
+  },
+  async getBenefitsRankingStatistics() {
```

Comment:
Async pas nécessaire ici
```suggestion
 getBenefitsRankingStatistics() {
```